src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -1455,6 +1502,13 @@ void uct_dc_mlx5_ep_handle_failure(uct_dc_mlx5_ep_t *ep, void *arg,
          * uct_dc_mlx5_iface_progress_pending call to avoid reset of working
          * DCI */
         uct_dc_mlx5_iface_reset_dci(iface, dci_index);
+
+        if (ep == iface->tx.fc_ep) {
```

Comment:
maybe instead of this, we could do if (ep!=fc_ep) before line 1496?

Reply:
it was the first idea, but it doesn't work.
we should allow progressing of pending requests after DCI was released (i.e. it means that all data were sent)